lib/src/gesture_recognizers.dart

```diff
@@ -0,0 +1,55 @@
+part of 'hooks.dart';
+
+abstract class _GestureRecognizerHook<G extends GestureRecognizer>
```

Comment:
Generic hook class is to make creating gesture detectors hooks involve as small code duplication as possible.